documentation/specs/Toast.md

```diff
@@ -0,0 +1,174 @@
+    div.setAttribute("style", "position: fixed; top: 0%; right: 50%");
+    document.getElementsByTagName("body")[0].prepend(div);
+
+    setIsReady(true);
+
+    //Clean up
+    return () => {
+      document.getElementsByTagName("body")[0].removeChild(div);
+    };
+  }, [containerId]);
+
+  return { isReady, containerId };
+};
+
+// Hook
+export const useToastDismiss = ({ toasts, setToasts, duration }) => {
+  const [dismissing, setDismissing] = useState("");
+
+  useEffect(() => {
+    if (dismissing) {
+      setToasts((t) => t.filter((_t) => _t.id !== dismissing));
+    }
+  }, [dismissing, setToasts]);
+
+  useEffect(() => {
+    if (toasts.length) {
+      const id = toasts[toasts.length - 1].id;
+      setTimeout(() => setDismissing(id), duration);
+    }
+  }, [toasts, duration]);
+};
+
+// ToastList, typing omitted
+export const ToastList = forwardRef(({ duration = 4000 }, ref) => {
+  const [toasts, setToasts] = useState([]);
+  const { isReady, containerId } = useToastContainer();
+
+  useToastDismiss({
+    toasts,
+    setToasts,
+    duration,
+  });
+
+  useImperativeHandle(ref, () => ({
+    queue(toast) {
+      setToasts([...toasts, { ...toast, id: uniqueIdGenerator() }]);
+    },
+  }));
+
+  return isReady
+    ? ReactDOM.createPortal(
+        <div style={{ display: "flex", gap: "12px", flexDirection: "column" }}>
+          {toasts.map((t) => (
+            <Toast key={t.id} status={t.status}>
+              {t.message}
+            </Toast>
+          ))}
+        </div>,
+
+        document.getElementById(containerId),
+      )
+    : null;
+});
+```
+
+To use, we now just need to import `ToastList` somewhere in our application where we want to display a toast and we can queue a message:
+
+```tsx
+import Button from "@easypost/easy-ui/Button";
+import ToastList from "a-place-in-your-app";
+import { useRef } from "react";
+
+function Component() {
+  const toastRef = useRef();
+
+  const handleSignUp = () => {
+    // inform user of successful account creation
+    toastRef.current.queue({ message: "Account Created", status: "success" });
+  };
+
+  return (
+    <>
+      <Button onPress={handleSignUp}>Sign up</Button>
+      <ToastList ref={toastRef} />
+    </>
+  );
+}
+```
+
+## Behavior
+
+### Accessibility
+
+The `Toast` component has an ARIA `role="status"` and elements with the role status have an implicit `aria-live` value of `polite` and an implicit `aria-atomic` value of `true`. Toasts are **not** suitable for dynamically changing content, consider using the `AlertBanner` component for such a use case.
+
+The `Toast` component does not provide a way for the user to manually dismiss the message, hence toasts should timeout after some time. When making decisions on the timeout duration, consider that users have different reading speeds, vision levels, and literacy levels. Avoid messages longer than a sentence and avoid rendering interactive elements within toasts that require the user to act.
```

Comment:
i noticed some libraries have special messages relating to toasts that show errors. like discouraging automatic dismissal or increasing the automatic dismissal duration. out of curiosity, do we need any special consideration for error toasts or is it the same as other toasts?